cratedb-demo/src/conf.py

```diff
@@ -0,0 +1,34 @@
+from transformers import AutoImageProcessor, AutoModel, AutoTokenizer
+from crate import client
+class ModelConf():
+    def __init__(self,device="cpu") -> None:
+        self.device=device
+    
+    def getModel(self):
+        return AutoModel.from_pretrained("openai/clip-vit-base-patch16").to(self.device)
+
+    def getProcessor(self):
+        return AutoImageProcessor.from_pretrained("openai/clip-vit-base-patch16")
+    
+    def getTokenizer(self):
+        return AutoTokenizer.from_pretrained("openai/clip-vit-base-patch16")
+    
+    def getModelConfImg(self):
+        model = self.getModel()
+        processor = self.getProcessor()
+        return model,processor
+    
+    def getModelConfText(self):
+        model = self.getModel()
+        tokenizer = self.getTokenizer()
+        return model,tokenizer
```

Comment:
I'm not sure why model and tokeniser etc are methods rather than just set in __init__ .. is it because you want to create them lazily ? 

perhaps you can refactor it to :

```python
class ModelConf:
    def __init__(self, device="cpu") -> None:
        self.model = AutoModel.from_pretrained("openai/clip-vit-base-patch16").to(device)
        self.processor = AutoImageProcessor.from_pretrained("openai/clip-vit-base-patch16")
        self.tokenizer = AutoTokenizer.from_pretrained("openai/clip-vit-base-patch16")

```

Reply:
Yes, I wanted to only create them when needed. But I have refactored them as per your suggestion as it might be a better approach